clang/test/SemaHLSL/BuiltIns/fmod-errors.hlsl

```diff
@@ -0,0 +1,38 @@
+
+// RUN: %clang_cc1 -finclude-default-header -triple dxil-pc-shadermodel6.6-library %s -fnative-half-type -emit-llvm-only -disable-llvm-passes -verify -verify-ignore-unexpected
+
+float test_too_few_arg() {
+  return __builtin_elementwise_fmod();
+  // expected-error@-1 {{too few arguments to function call, expected 2, have 0}}
+}
+
+float2 test_too_many_arg(float2 p0, float2 p1, float2 p3) {
+  return __builtin_elementwise_fmod(p0, p1, p3);
+  // expected-error@-1 {{too many arguments to function call, expected 2, have 3}}
+}
+
+float builtin_bool_to_float_type_promotion(bool p1, bool p2) {
+  return __builtin_elementwise_fmod(p1, p2);
+  // expected-error@-1 {{passing 'bool' to parameter of incompatible type 'float'}}
+}
+
+float builtin_fmod_int_to_float_promotion(int p1, int p2) {
+  return __builtin_elementwise_fmod(p1, p2);
+  // expected-error@-1 {{passing 'int' to parameter of incompatible type 'float'}}
+}
+
+float2 builtin_fmod_int2_to_float2_promotion(int2 p1, int2 p2) {
+  return __builtin_elementwise_fmod(p1, p2);
+  // expected-error@-1 {{passing 'int2' (aka 'vector<int, 2>') to parameter of incompatible type '__attribute__((__vector_size__(2 * sizeof(float)))) float' (vector of 2 'float' values)}}
+}
+
+// builtins are variadic functions and so are subject to DefaultVariadicArgumentPromotion
+half builtin_fmod_half_scalar (half p0, half p1) {
```

Comment:
This doesn’t look right.  Which makes me think we shouldn’t capture this implementation detail if  fmod hlsl api is working fine with float/half. You are passing half/float in these two cases. For cases like this future folks are going to want to change/fix this test case. To avoid the confusion caused by variadic argument promotion we should care more what the fmod sema error checking returns.

Reply:
@farzonl You mean that we should remove builtin_fmod_half_scalar and builtin_fmod_float_scalar tests?

Reply:
Yeah I don't like these tests because we don't really case about the builtin behavior  we care about the HLSL api's behavior and in the HLSL api we pass the type to the builtin and we don't end up in the `DefaultVariadicArgumentPromotion` case.

Reply:
Done. Removed builtin_fmod_half_scalar and builtin_fmod_float_scalar tests.